src/main/java/com/nextcloud/android/sso/api/AidlNetworkRequest.java

```diff
@@ -164,16 +163,21 @@ private void waitForApi() throws NextcloudApiNotRespondingException {
     public Response performNetworkRequestV2(NextcloudRequest request, InputStream requestBodyInputStream) throws Exception {
         ParcelFileDescriptor output = performAidlNetworkRequestV2(request, requestBodyInputStream);
         InputStream os = new ParcelFileDescriptor.AutoCloseInputStream(output);
-        ExceptionResponse response = deserializeObjectV2(os);
+        try {
+            ExceptionResponse response = deserializeObjectV2(os);
 
-        // Handle Remote Exceptions
-        if (response.getException() != null) {
-            if (response.getException().getMessage() != null) {
-                throw parseNextcloudCustomException(response.getException());
+            // Handle Remote Exceptions
+            if (response.getException() != null) {
+                if (response.getException().getMessage() != null) {
+                    throw parseNextcloudCustomException(response.getException());
+                }
+                throw response.getException();
             }
-            throw response.getException();
+            return new Response(os, response.headers);
+        } catch (Exception e) {
+            os.close();
+            throw e;
```

Comment:
Don't like this...I'm open to suggestions

Reply:
Remove catch and put os.close() in finally?
Then Exception will be thrown, but also os.close() is called?

(not tested, but should work)

Reply:
That was my initial solution. Then I noticed `os` is returned and is intended to stay open, it only needs to be closed on an exception when it isn't returned anymore.

Reply:
Ah. Then I think the solution is good.
Can you add an info that os normally should stay open, but only be closed when handling an Exception?

Reply:
Something like [this](https://github.com/nextcloud/Android-SingleSignOn/compare/e1b80f18df5b453ea2367937fe004ba0ebfbae87..4b1a3cb821f61b3ac604b54a102efcf5ef04a6ca)?